posts/2020-11-30-open-liberty-devfile-stack.adoc

```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title: "Develop Open Liberty apps directly in OpenShift"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/awisniew90
+author_github: https://github.com/awiniew90
+seo-title: Develop Open Liberty apps directly in OpenShift - OpenLiberty.io
+seo-description: Get true-to-production development directly in a Kubernetes / OpenShift cluster
+blog_description: "Get true-to-production development directly in a Kubernetes / OpenShift cluster"
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Develop Open Liberty apps directly in OpenShift
+Adam Wisniewski <https://github.com/awisniew90>
+
+True-to-production development is critical when creating cloud applications, and what better way to develop those apps than directly in the cloud! The link:https://github.com/OpenLiberty/application-stack[Java Open Liberty devfile stack] allows you to do just that by providing a simple yet robust Open Liberty development experience directly in a Kubernetes / OpenShift cluster. It does the heavy lifting allowing you to focus on your code as if it were running right on your laptop. 
+
+== OpenShift Do (ODO)
+
+link:https://odo.dev/[ODO] is the mechanism used to drive the Open Liberty stack. It is a straightforward CLI tool that allows you to create stack projects and interact directly with a Kubernetes / OpenShift cluster. Unlike `kubectl` and `oc` it abstracts away the complexity of the underlying infrastructure and helps developers focus on their applications. 
+
+== Try it out!
+
+After link:https://odo.dev/docs/installing-odo/[installing ODO], you can use the Open Liberty stack to either link:https://github.com/OpenLiberty/application-stack/wiki/Using-the-Default-Starter#getting-started[create a new Open Liberty starter application] or initialize an existing app. For this blog, we will be using the link:https://github.com/OpenLiberty/application-stack-intro[application-stack-intro] project as a starting point.
+
+
+[source,sh]
+----
+$ git clone git@github.com:OpenLiberty/application-stack-intro.git && \
+  cd application-stack-intro
+----
+
+This application is a basic JAX-RS sample using Maven, Open Liberty and Microprofile. 
+You will also notice an additional `devfile` packaged with the app. This is the brains behind the Open Liberty stack and outlines how the application will be built, run and deployed using the stack's development image. 
+
+For convenience and illustration, this devfile is included with the intro app; however, there is nothing app specific in the devfile. If one was not included, the same devfile would be automatically pulled down from the odo devfile registry when initialized with odo. No changes necessary. 
```

Comment:
In general there might be app-specific tweaks.   If one were not included you could easily use the one from the devfile registry with the extra 'java-openliberty' arg on `odo create`.    

I feel like saying:
> the same devfile would be automatically pulled down 
confuses a bit as to how this all works.